aiohttp/web_protocol.py

```diff
@@ -523,24 +523,28 @@ def handle_error(self,
         information. It always closes current connection."""
         self.log_exception("Error handling request", exc_info=exc)
 
+        ct = 'text/plain'
         if status == 500:
-            msg = "<h1>500 Internal Server Error</h1>"
-            if self.debug:
-                with suppress(Exception):
-                    tb = traceback.format_exc()
-                    tb = html_escape(tb)
-                    msg += '<br><h2>Traceback:</h2>\n<pre>'
-                    msg += tb
-                    msg += '</pre>'
+            msg = 'Server got itself in trouble'
+            if 'text/html' in request.headers.get('Accept', ''):
+                if self.debug:
+                    with suppress(Exception):
+                        tb = html_escape(traceback.format_exc())
+                        msg = '<h2>Traceback:</h2>\n<pre>{}</pre>'.format(tb)
+                message = (
+                    "<html><head>"
+                    "<title>500 Internal Server Error</title>"
+                    "</head><body>\n<h1>500 Internal Server Error</h1>"
+                    "<br>\n{msg}\n</body></html>\n"
+                ).format(msg=msg)
+                ct = 'text/html'
             else:
-                msg += "Server got itself in trouble"
-                msg = ("<html><head><title>500 Internal Server Error</title>"
-                       "</head><body>" + msg + "</body></html>")
-            resp = Response(status=status, text=msg, content_type='text/html')
-        else:
-            resp = Response(status=status, text=message,
-                            content_type='text/html')
+                if self.debug:
+                    with suppress(Exception):
+                        msg = traceback.format_exc()
+                message = '500 Internal Server Error\n\n' + msg
```

Comment:
You're not reusing `msg = 'Server got itself in trouble'` from above so messages are not consistent depending on the content-type. I'd probably put it into a separate var like `msg_subtitle`